sdk/eventgrid/azure-messaging-eventgrid/migration-guide.md

```diff
@@ -0,0 +1,174 @@
+- A unified asynchronous programming model using [Project Reactor][project-reactor].
+- A unified method of creating clients via client builders to interact with Azure services.
+
+## Important changes
+
+### Group id, artifact id, and package names
+
+Group ids, artifact ids, and package names for the modern Azure client libraries for Java have changed. They follow the [Java SDK naming guidelines][GuidelinesJavaDesign]. Each will have the group id `com.azure`, an artifact id following the pattern `azure-[area]-[service]`, and the root package name `com.azure.[area].[Service]`. The legacy clients have a group id `com.microsoft.azure` and their package names followed the pattern `com.microsoft. Azure.[service]`. This provides a quick and accessible means to help understand, at a glance, whether you are using modern or legacy clients.
+
+In EventGrid, the modern client libraries have packages and namespaces that begin with `com.azure.messaging.eventgrid` and were released starting with 4.0.0. The legacy client libraries have package names starting with `com.microsoft.azure.eventgrid` and a version of 1.x.x or below.
+The new SDK starts with 4.0.0 to be inline with EventGrid SDKs of other languages.
+
+#### Instantiating clients
+
+In 1.x, the `EventGridClient` was instantiated via the `EventGridClientImpl` constructor. The client contained both sync and async methods.
+The client can send only `EventGridEvent` to an EventGrid topic or domain.
+```java
+String endpoint = "<endpont of your event grid topic/domain that accepts EventGridEvent schema>";
+TopicCredentials topicCredentials = new TopicCredentials(key);
+EventGridClient client = new EventGridClientImpl(topicCredentials);
+```
+
+In 4.x, the creation of the client is done through the [EventGridPublisherClientBuilder][EventGridPublisherClientBuilder]. The sync and async operations are separated to [EventGridPublisherClient] and [EventGridPublisherAsyncClient].
+```java
+EventGridPublisherClient<EventGridEvent> eventGridEventClient = new EventGridPublisherClientBuilder()
+    .endpoint("<endpont of your event grid topic/domain that accepts EventGridEvent schema>")
+    .credential(new AzureKeyCredential("<key for the endpoint>"))
+    .buildEventGridEventPublisherClient();
+```
+Aside from [EventGridEvent][EventGridEvent], the generic client can be instantiated to also send `CloudEvent`, or custom events to an EventGrid topic or domain.
+Refer to the [README][README] for more information about sending `CloudEvent` and custom events.
+
+#### Send events to an Event Grid Topic
+In 1.x, the `publishEvent` method has the EventGrid endpoint as a parameter.
+```java
+List<EventGridEvent> eventsList = new ArrayList<>();
+for (int i = 0; i < 5; i++) {
+    eventsList.add(new EventGridEvent(
+        UUID.randomUUID().toString(),
+        String.format("Door%d", i),
+        new ContosoItemReceivedEventData("Contoso Item SKU #1"),
+        "Contoso.Items.ItemReceived",
+        DateTime.now(),
```

Comment:
Setting the datetime is not shown in the 4.x sample. We should note that it defaults to current time if not set or we should show how to set it.

Reply:
Added a sentence for default values.